lib/cloudregister/registerutils.py

```diff
@@ -2027,6 +2034,42 @@ def has_network_access_by_ip_address(server_ip):
     return True
 
 
+# ----------------------------------------------------------------------------
+def is_suma_instance():
+    """Return True if there is a match for SUMA and SLE-Micro products,
+    otherwise False."""
+    products = glob.glob('/etc/products.d/*.prod')
+    num_matches = 0
+    suma_prods = [
+        '/etc/products.d/suse-manager-server.prod',
+        '/etc/products.d/sle-micro.prod'
+    ]
+    for product in products:
+        if product.lower() in suma_prods:
+            num_matches += 1
+
+    return num_matches == len(suma_prods)
+
+
+# ----------------------------------------------------------------------------
+def get_suma_registry_content():
+    """Return the content of the SUMA_REGISTRY_CONF_PATH."""
+    suma_reg_config_data = {}
+    try:
+        with open(SUMA_REGISTRY_CONF_PATH, 'r') as suma_reg_config:
+            suma_reg_config_data = yaml.safe_load(suma_reg_config)
+        return suma_reg_config_data, False
+    except IOError as error:
+        logging.info(str(error))
+        action = 'open'
+    except yaml.YAMLError:
+        action = 'parse'
+
+    logging.info('Could not %s %s' % (action, SUMA_REGISTRY_CONF_PATH))
+    failed = __mv_file_backup(SUMA_REGISTRY_CONF_PATH)
```

Comment:
I'm not 100 % sure we should move the file to a backup, that would make `cloud-regionsrv-client` the owner of that file and IMO it should be owned by SUMA

that would mean we log that file could not be opened or parsed and no further action

Reply:
I agree, the backup strategy in this case might be unexpected. Just the logging is imho enough